### PR TITLE
Cancel hearing page

### PR DESCRIPTION
**Pull request by Kartheek-S**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4467

### Change description ###
Cancel hearing

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: api/hearings/data/reference.mock.data.ts [hunk 1]
```diff
@@ -165,4 +198,8 @@ export const ALL_REF_DATA: RefDataByCategoryModel[] = [
     categoryKey: 'JudgeType',
     services: [SSCS_JUDGE_TYPES],
   },
+  {
+    categoryKey: 'CancelHearing',
```
Comment (rayliangatos) [inline api/hearings/data/reference.mock.data.ts:202]: can this categoryKey change to be 'CancelHearingReason'

File: api/hearings/data/reference.mock.data.ts [hunk 2]
```diff
@@ -128,6 +128,35 @@ export const DEFAULT_PARTYCHANNEL_REF: RefDataModel[] = [
     parentKey: null,
   },
 ];
+export const DEFAULT_HEARING_CANCEL_REF: RefDataModel[] = [
+  {
+    key: 'reasoneone',
```
Comment (rayliangatos) [inline api/hearings/data/reference.mock.data.ts:133]: typo, can this key be reasonOne?

File: api/hearings/data/reference.mock.data.ts [hunk 3]
```diff
@@ -128,6 +128,35 @@ export const DEFAULT_PARTYCHANNEL_REF: RefDataModel[] = [
...
+    value_en: 'Reason 1',
+    value_cy: '',
+    hintText_EN: 'reason 1',
+    hintTextCY: '',
+    order: 1,
+    parentKey: null,
+  },
+  {
+    key: 'reasonetwo',
```
Comment (rayliangatos) [inline api/hearings/data/reference.mock.data.ts:142]: typo, can this key be reasonTwo?

File: api/hearings/data/reference.mock.data.ts [hunk 4]
```diff
@@ -128,6 +128,35 @@ export const DEFAULT_PARTYCHANNEL_REF: RefDataModel[] = [
...
+    value_en: 'Reason 2',
+    value_cy: '',
+    hintText_EN: 'Reason 2',
+    hintTextCY: '',
+    order: 2,
+    parentKey: null,
+  },
+  {
+    key: 'reasonthree',
```
Comment (rayliangatos) [inline api/hearings/data/reference.mock.data.ts:151]: typo, can this key be reasonThree?

File: src/hearings/containers/cancel-hearing/cancel-hearing.component.html [hunk 5]
```diff
@@ -2,9 +2,43 @@
...
+            </h1>
+            <exui-hearing-parties-title></exui-hearing-parties-title>
+            <div class="govuk-form-group">
+              <fieldset class="govuk-fieldset">
+                  <legend class="govuk-fieldset__legend govuk-fieldset__legend--s">
+                      <h1 class="govuk-fieldset__heading">
+                          Why is this hearing being cancelled?
+                      </h1>
+                  </legend>
+                  <div id="other-panel-roles-hint" class="govuk-hint">
```
Comment (rayliangatos) [inline src/hearings/containers/cancel-hearing/cancel-hearing.component.html:23]: is this panel? the name of id exactly matched what this page?

File: src/hearings/containers/cancel-hearing/cancel-hearing.component.ts [hunk 6]
```diff
@@ -1,15 +1,57 @@
...
+  }
+
+  public isFormValid(): boolean {
+    return this.hearingCancelForm.valid;
+  }
+
+  public checkFormData(): void {
+    this.hearingCancelSelectionError = null;
+    this.validationErrors = [];
+    // TODO: check form data
```
Comment (rayliangatos) [inline src/hearings/containers/cancel-hearing/cancel-hearing.component.ts:55]: can this be removed?

File: src/hearings/models/refData.model.ts [hunk 7]
```diff
@@ -6,6 +6,7 @@ export interface RefDataModel {
   hintTextCY: string;
   order: number;
   parentKey: string;
+  selected?: boolean;
```
Comment (rayliangatos) [inline src/hearings/models/refData.model.ts:9]: this is a generic model from refdata API, what is the purpose of adding this selected?

File: src/hearings/containers/cancel-hearing/cancel-hearing.component.ts (hunk 6 above)
Comment (Kartheek-S) [reply src/hearings/containers/cancel-hearing/cancel-hearing.component.ts]: removed

File: src/hearings/containers/cancel-hearing/cancel-hearing.component.html (hunk 5 above)
Comment (Kartheek-S) [reply src/hearings/containers/cancel-hearing/cancel-hearing.component.html]: removed id. Sorry, its copied from mockup

File: src/hearings/models/refData.model.ts (hunk 7 above)
Comment (Kartheek-S) [reply src/hearings/models/refData.model.ts:9]: We need to know which are selected if this model is used as checkboxes. So, added it as optional

**Review by rayliangatos — APPROVED** [review]

**Review by waynepHMCTS — APPROVED** [review]

**Pull request merged**